crates/ra_lsp_server/tests/heavy_tests/main.rs

```diff
@@ -1,17 +1,48 @@
 mod support;
 
+use std::{
+    collections::HashMap,
+    time::Instant,
+};
+
 use languageserver_types::{
     CodeActionContext, DocumentFormattingParams, FormattingOptions, Position, Range,
 };
 use ra_lsp_server::req::{
-    CodeActionParams, CodeActionRequest, Formatting, Runnables, RunnablesParams,
+    CodeActionParams, CodeActionRequest, Formatting, Runnables, RunnablesParams, CompletionParams, Completion,
 };
 use serde_json::json;
 
 use crate::support::project;
 
 const LOG: &'static str = "";
 
+#[test]
+fn completes_items_from_standard_library() {
+    let project_start = Instant::now();
+    let server = project(
+        r#"
+//- Cargo.toml
+[package]
+name = "foo"
+version = "0.0.0"
+
+//- src/lib.rs
+use std::collections::Spam;
+"#,
+    );
+    server.wait_for_feedback("workspace loaded");
+    eprintln!("loading took    {:?}", project_start.elapsed());
+    let completion_start = Instant::now();
+    let res = server.send_request::<Completion>(CompletionParams {
+        text_document: server.doc_id("src/lib.rs"),
+        context: None,
+        position: Position::new(0, 23),
+    });
+    assert!(format!("{}", res).contains("HashMap"));
+    eprintln!("completion took {:?}", completion_start.elapsed());
```

Comment:
For **debug** build I get

```
loading took    4.014003867s
completion took 6.368790446s
```

For release, 

```
loading took    303.160112ms
completion took 321.775276ms
```

Reply:
That completion time seems unfortunately long.

Reply:
That's a one-time from scratch analysis. All subsequent completions are instant.

What we need to do ideally though is to kick basic analysis eagerly, after we've loaded the workspace, and not lazily on completion...